### ISSUE TITLE
German localization and some bug fixes

### ISSUE THREAD

**Issue by Tekl** [Stephen-Lon/Alfred-workflow-open-macos-13-settings-panes#11]
Hi there,

I've created a German localization of the "shell script" and also fixed some paths and URLs in the English version. Feel free to integrate it.

Cheers

[syset.zip](https://github.com/user-attachments/files/18065000/syset.zip)

**Comment by Stephen-Lon** [comment]
I apologise for the delay in responding but am very busy just at the moment. However, I appreciate your kind contribution and will look at it as soon as I get time.

I do have two basic questions:

1. How will the German localisation be picked up when someone has set German as their language in macOS. (There may be something basic I'm not understanding—sorry if so!)
2. How will the German localisation be maintained promptly when I post updates to the original? My knowledge of your language is a little basic!

Aber, vielen dank!

Stephen

**Comment by Tekl** [comment]
Currently, I just change the script file name in the Script Filter. But it could be enhanced like this:

`./syset_${language}.sh`

Then you can add a Language Popup in the Configuarion Builder. The scripts should be named `syset_en.sh` and `syset_de.sh`. 

I don't know how to get it maintained well. Maybe you can just publish the outdated German version and wait until someone (if not me) will merge the changes from the English version into the German version.

**Comment by Stephen-Lon** [comment]
You have been extremely helpful and your idea for selecting the language works perfectly. I have implemented that. I've also substantially updated the JSON (English version) following your comments. I'm afraid some of the changes will require more work on your part with the German localisation: sorry!

I attach a zip file containing the new English version, a note of the changes I've made and the updated workflow (which, of course, currently uses your old localisation).

Many thanks again for your help and suggestions.
[Archive.zip](https://github.com/user-attachments/files/18125753/Archive.zip)

Stephen

**Comment by Tekl** [comment]
No problem, I have changed the localization. Is there a reason, why stuff like sharing settings, software updates and transfer/reset are is not under General? I've moved them in the English version as well and corrected two URLs. I also added a note with the changes.
[Archiv.zip](https://github.com/user-attachments/files/18147134/Archiv.zip)

**Comment by Stephen-Lon** [comment]
Thanks again for all your work and I'm sorry for not being so thorough as I thought I usually am!

It all looks good now save for one very strange problem that affects three changes you made. General → Sharing → Remote Management, General → Sharing → Screen Sharing & General → Time Machine. In each of those cases the only way I can get the relevant item to display in Alfred is to use your "match" word. E.g., "syset time" or "syset machine" will _not_ display General → Time Machine…only "backup” will. Do you have any idea why?

Stephen

**Comment by Stephen-Lon** [comment]
I omitted to mention that in the General → Sharing → Remote Application Scripting entry you need to update the other fields with the new title. (I have done so in the en version on which I'm working.)

**Comment by Stephen-Lon** [comment]
Currently resolving the matching issue by including in the "match:” you added to each the other relevant words.

**Comment by Stephen-Lon** [comment]
With many thanks again for all your work here is the latest syset_en.sh. Aside from moving some items into a sort of alphabetical order (which makes the JSON slightly easier for me to maintain!) I have:

1.  Made the relevant changes to the "match:” fields in General → Sharing → Remote Management, General → Sharing → Screen Sharing & General → Time Machine.
2.  Included the small corrections in the General → Sharing → Remote Application Scripting appropriate fields.

Do you think we are finally "done”? 😁 
[syset_en.sh.zip](https://github.com/user-attachments/files/18149732/syset_en.sh.zip)

**Comment by Stephen-Lon** [comment]
Hmmm, maybe not done… Although when I change the language to German and run `syset` in Alfred the debugger shows the German JSON without error no German (or anything else) is shown in Alfred.

**Comment by Tekl** [comment]
It appears that match will override the title completely. I also added the relevant words to the match field in the attached german version. I fixed a missing comma as Alfred didn't show any results.

Did you mean "Remote Login" instead of "Remote Application Scripting" for updating the other fields? It's the only place where I can find it.

"Date & Time" should also be sorted into "General". I already did this in the German version.

I also added lockdown mode "x-apple.systempreferences:com.apple.preference.security?LockdownMode" and App Management "x-apple.systempreferences:com.apple.preference.security?Privacy_AppBundles".

[syset_de.sh.zip](https://github.com/user-attachments/files/18150103/syset_de.sh.zip)

**Comment by Tekl** [comment]
Oops, I missed some keys for "App Managment". Fixed. Sorry.
[syset_de.sh.zip](https://github.com/user-attachments/files/18150169/syset_de.sh.zip)

**Comment by Stephen-Lon** [comment]
I agree with what you say about the "match" field and have updated a number of those in the English version (I don't think those updates need to trouble you for the German version). 

Your latest version works correctly when the language is set to German, thanks. (I  _hate_ JSON and its commas!)

I've put "Date & Time” under "General → Date & Time” now and have also added lockdown mode and app management (thanks).

I'm slightly tempted to bump the version number a little higher (3.5?) in view of all the changes. What do you think?

Thanks again for all your patient help.

Latest English version attached.
[syset_en 2.sh.zip](https://github.com/user-attachments/files/18150897/syset_en.2.sh.zip)

**Comment by Tekl** [comment]
It appears to be fine now. The version bump would be a good fit for these enhancements. Maybe we could add some missing Settings from [this list](https://github.com/bvanpeski/SystemPreferences/blob/main/macos_preferencepanes-Ventura.md), which I've discovered recently.

**Comment by Stephen-Lon** [comment]
I'll bump the version number to 3.5. I've been following [this GitHub site](https://gist.github.com/rmcdongit/f66ff91e0dad78d4d6346a75ded4b751) for links. It's reasonably well up-to-date for macOS 15.

Let's get what we've done released now as there are significant changes. We can always add at leisure in due course.

Thanks again—you're a star!
